From 1b3a455f131f0359f52267e47fde6cfbad533a4a Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Tue, 25 Apr 2023 18:05:18 +0200
Subject: [PATCH 517/954] feat(stm32mp25): put PKG numbers under flag

The packages will be different on STM32MP2 variants. Put STM32MP25 package
identifiers under STM32MP25 flag.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 14ba03c13a325cd570d6ef9c31438c2dd6b69581 feat(stm32mp25): put PKG numbers under flag]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: I1c5c19dd07ef1b3a140fc24e26cd53bbbd31a7b1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/356307
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h     | 2 ++
 plat/st/stm32mp2/stm32mp2_private.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 130be2591f..70da89b755 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -115,11 +115,13 @@
 /*******************************************************************************
  * PACKAGE ID
  ******************************************************************************/
+#if STM32MP25
 #define STM32MP25_PKG_CUSTOM		U(0)
 #define STM32MP25_PKG_AL_TBGA361	U(3)
 #define STM32MP25_PKG_AK_TBGA424	U(4)
 #define STM32MP25_PKG_AI_TBGA436	U(5)
 #define STM32MP25_PKG_UNKNOWN		U(7)
+#endif /* STM32MP25 */
 
 /*******************************************************************************
  * STM32MP2 memory map related constants
diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index 0823f3a9f6..8320b0d710 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -371,6 +371,7 @@ void stm32mp_get_soc_name(char name[STM32_SOC_NAME_SIZE])
 
 	/* Package */
 	switch (get_cpu_package()) {
+#if STM32MP25
 	case STM32MP25_PKG_CUSTOM:
 		pkg = "XX";
 		break;
@@ -383,6 +384,7 @@ void stm32mp_get_soc_name(char name[STM32_SOC_NAME_SIZE])
 	case STM32MP25_PKG_AI_TBGA436:
 		pkg = "AI";
 		break;
+#endif /* STM32MP25 */
 	default:
 		pkg = "??";
 		break;
-- 
2.53.0

